doc/read-the-docs-site/howtos/exporting-a-blueprint.rst

```diff
@@ -0,0 +1,403 @@
+.. highlight:: haskell
+.. _exporting_a_blueprint:
+
+How to produce a Plutus Contract Blueprint
+==========================================
+
+Plutus Contract Blueprints (`CIP-0057`_) are used to document the binary interface of a
+Plutus contract in a machine-readable format (JSON schema).
+
+A contract Blueprint can be produced by using the
+`writeBlueprint` function exported by the `PlutusTx.Blueprint` module::
+
+  writeBlueprint
+    :: FilePath
+    -- ^ The file path where the blueprint will be written to,
+    --  e.g. '/tmp/plutus.json'
+    -> ContractBlueprint
+    -- ^ Contains all the necessary information to generate 
+    -- a blueprint for a Plutus contract.
+    -> IO ()
+
+In order to demonstrate the usage of the `writeBlueprint` function,
+Let's consider the following example validator function and its interface:
+
+.. literalinclude:: Cip57Blueprint.hs
+   :start-after: BEGIN interface types
+   :end-before:  END interface types
+
+.. literalinclude:: Cip57Blueprint.hs
+   :start-after: BEGIN validator
+   :end-before:  END validator
+
+First of all we need to import required functionality:
```

Comment:
personally I try to go through the blocks in an order that it is at least possible to put them in a file in. Yes, anyone who knows Haskell will no think that they can put the imports in after some data declarations... but I still think it reads a little confusingly.